File: S/908-smallest-range-i/main.go
package main

import "fmt"

func smallestRangeI(A []int, K int) int {
	min, max := 1<<31-1, -1
	for i := 0; i < len(A); i++ {
		if min > A[i] {
			min = A[i]
		}
		if max < A[i] {
			max = A[i]
		}
	}
	if max-min-2*K <= 0 {
		return 0
	}

	return max - min - 2*K
}

func main() {
	fmt.Println("a")
}

File: S/908-smallest-range-i/main.rs
struct Solution {}

impl Solution {
  pub fn smallest_range_i(nums: Vec<i32>, k: i32) -> i32 {
    let (max, min) = nums.iter().fold((i32::MIN, i32::MAX), |(max, min), &v| {
      (max.max(v), min.min(v))
    });
    if max - min >= 2 * k {
      max - min - 2 * k
    } else {
      0
    }
  }
}
